app/models/mark.rb

```diff
@@ -0,0 +1,102 @@
+
+class Mark < ActiveRecord::Base
+  has_and_belongs_to_many :spies
+  has_many :media
+
+  validates :username, :provider, :uid, presence: true
+  validates :uid, uniqueness: true
+
+  def self.vimeo_lookup(search_term)
+    auth = "Bearer #{ENV['VIMEO_ACCESS_TOKEN']}"
+    marks = HTTParty.get("https://api.vimeo.com/users", query: {:query => search_term}, headers: { "Authorization" => auth })
+    parsed_marks = JSON.parse(marks)
+    marks_data = parsed_marks["data"]
+    result = []
+    marks_data.each do |mark|
+      unless mark["pictures"].nil?
+        image = mark["pictures"]["sizes"].last
+      end
+
+      if image.nil?
+        profile_image = "blank.png"
+      else
+        profile_image = image["link"]
+      end
+
+      uid = mark["uri"].gsub(/[^\d]/, '')
+      username = mark["link"].split('/')[-1]
+
+      result <<  Mark.new(
+        username: username,
+        name: mark["name"],
+        bio: mark["bio"],
+        location: mark["location"],
+        link: mark["link"],
+        image_url: profile_image,
+        uid: uid,
+        provider: "vimeo"
+        )
+    end
+    return result
+  end
+
+  def self.single_mark_vimeo_lookup(search_term)
+    auth = "Bearer #{ENV['VIMEO_ACCESS_TOKEN']}"
+    mark = HTTParty.get("https://api.vimeo.com/users/#{search_term}/", headers: { "Authorization" => auth })
+    mark_parsed = JSON.parse(mark)
+
+    if !mark_parsed["error"].nil?
+      return mark_parsed["error"]
+    else
+      unless mark_parsed["pictures"].nil?
+        image = mark_parsed["pictures"]["sizes"].last
+      end
+
+      if image.nil?
+        profile_image = "blank.png"
+      else
+        profile_image = image["link"]
+      end
+
+      uid = mark_parsed["uri"].gsub(/[^\d]/, '')
+
+      result =  Mark.new(
+        username: search_term,
+        name: mark_parsed["name"],
+        bio: mark_parsed["bio"],
+        location: mark_parsed["location"],
+        link: mark_parsed["link"],
+        image_url: profile_image,
+        uid: uid,
+        provider: "vimeo"
+        )
+    end
+    return result
+  end
+
+  def twitter
+    Seemore::Application.config.twitter
```

Comment:
You used this in the Marks controller as well -- can you think of a location to put this code so it isn't duplicated across your application?

Figuring out where to put all the funky extra methods necessary for an application was one of the hardest things for me when learning Rails, and I def still don't have it all figured out. If you keep thinking about (single responsibility principle)[https://en.wikipedia.org/wiki/Single_responsibility_principle] and (DRYing)[https://en.wikipedia.org/wiki/Don%27t_repeat_yourself] up your code, you'll develop a feeling for what goes where over time. Also, it can be largely opinion and style based, so take my advice with a bucket of salt. :wink: